File: dashboard/android/app/src/main/java/com/attobot/dashboard/ui/components/Pager.kt
package com.attobot.dashboard.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.attobot.dashboard.ui.theme.accent
import com.attobot.dashboard.ui.theme.border
import com.attobot.dashboard.ui.theme.muted
import com.attobot.dashboard.ui.theme.panel
import kotlin.math.ceil
import kotlin.math.max

/** Prev/next pager mirroring the RN <Pager/>. */
@Composable
fun Pager(
    offset: Int,
    limit: Int,
    total: Int,
    onPage: (Int) -> Unit,
) {
    val page = offset / limit + 1
    val pages = max(1, ceil(total.toDouble() / limit).toInt())
    Row(
        Modifier
            .fillMaxWidth()
            .padding(top = 14.dp),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        PagerButton("‹ prev", enabled = offset > 0) { onPage(max(0, offset - limit)) }
        Text(
            "page $page / $pages · $total total",
            color = muted,
            fontSize = 13.sp,
            modifier = Modifier.padding(horizontal = 14.dp),
        )
        PagerButton("next ›", enabled = offset + limit < total) { onPage(offset + limit) }
    }
}

@Composable
private fun PagerButton(
    label: String,
    enabled: Boolean,
    onClick: () -> Unit,
) {
    Surface(
        shape = RoundedCornerShape(6.dp),
        border = BorderStroke(1.dp, border),
        color = panel,
        modifier = Modifier.clickable(enabled = enabled, onClick = onClick),
    ) {
        Text(
            label,
            color = if (enabled) accent else muted,
            fontSize = 13.sp,
            modifier = Modifier.padding(horizontal = 12.dp, vertical = 6.dp),
        )
    }
}

File: dashboard/ios/OpenWorldDashboard/Views/Components/Pager.swift
import SwiftUI

/// Offset pager. "‹ prev" disabled at offset 0, "next ›" disabled at the end.
/// Mirrors the RN `Pager`.
struct Pager: View {
    let offset: Int
    let limit: Int
    let total: Int
    let onPage: (Int) -> Void

    var body: some View {
        let page = (offset / limit) + 1
        let pages = max(1, Int((Double(total) / Double(limit)).rounded(.up)))
        let prevDisabled = offset == 0
        let nextDisabled = offset + limit >= total

        HStack(spacing: 14) {
            pagerButton("‹ prev", enabled: !prevDisabled) {
                onPage(max(0, offset - limit))
            }
            Text("page \(page) / \(pages) · \(total) total")
                .font(.system(size: 13))
                .foregroundStyle(Theme.muted)
            pagerButton("next ›", enabled: !nextDisabled) {
                onPage(offset + limit)
            }
        }
        .padding(.top, 14)
        .frame(maxWidth: .infinity)
    }

    private func pagerButton(_ label: String, enabled: Bool, action: @escaping () -> Void) -> some View {
        Button(action: action) {
            Text(label)
                .font(.system(size: 13))
                .foregroundStyle(enabled ? Theme.accent : Theme.muted)
                .padding(.horizontal, 12)
                .padding(.vertical, 6)
                .background(Theme.panel)
                .overlay(
                    RoundedRectangle(cornerRadius: 6)
                        .stroke(Theme.border, lineWidth: 1)
                )
                .clipShape(RoundedRectangle(cornerRadius: 6))
        }
        .disabled(!enabled)
        .opacity(enabled ? 1.0 : 0.4)
    }
}
